bpdv3/Assets/Scripts/Floor/TileController.cs

```diff
@@ -25,15 +25,17 @@ void Start()
     // Update is called once per frame
     void Update()
     {
+        // Specified by the GDD, on the 6th beat the floor moves until it reaches the 12th beat. Seen below **.
         if (SceneController.instance.scrollBeatCount >= 6)
         {
             foreach (GameObject go in tilesArray)
             {
-                go.transform.Translate(direction.normalized * (Time.deltaTime * (1.25f / 4.0000005f)));
+                go.transform.Translate(direction.normalized * (Time.deltaTime * (1.25f / 4f)));
 
                 if (go.transform.position.z <= 0)
                 {
-                    go.GetComponent<TileProperties>().ResetValue();
+                    go.GetComponent<TileProperties>().ResetValue();         // Once a row has passed the last row hazard point (flames/laser). It's values can reset to zero.
```

Comment:
you dont have to get TileProperties on every update find it at Start() and assing it to some global variable

Reply:
And if this script has a dependency to TileProperties (this script crashes if TileProperties got deleted accidently), check this beauty https://docs.unity3d.com/ScriptReference/RequireComponent.html